### PR TITLE
Feature/simulation configuration

### PR DESCRIPTION
**Pull request by Florisheinen1**
This PR adds a networker class for publishing and subscribing to Simulation Configuration commands. These commands can edit the locations of robots in a simulator, edit the physical properties and limitations of the robots, change the position of the ball, change the simulation speed etc.

Even though this PR is not dependent on another PR, this would be a very useless PR if the [PR in RobotHub](https://github.com/RoboTeamTwente/roboteam_robothub/pull/49) would not be merged.

### CODE REVIEW

File: include/SimulationConfigurationNetworker.hpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include <proto/SimulationConfiguration.pb.h>
+#include <utils/Publisher.hpp>
+#include <utils/Subscriber.hpp>
+
+#include <functional>
```
Comment (emielsteerneman) [inline include/SimulationConfigurationNetworker.hpp:6]: Doesn't seem to be used

**Review by emielsteerneman — APPROVED** [review]
Tested it using roboteam_ai and it works fine. However, I'm wondering if it is simpler to just use the proto files that grSim accepts, instead of this intermediate format. If we ever use a different simulator, we can simply convert the grSim packets, instead of converting the intermediate packets.

Also, is there a possibility to add field configuration to this?

File: include/SimulationConfigurationNetworker.hpp (hunk 1 above)
Comment (Florisheinen1) [reply include/SimulationConfigurationNetworker.hpp:6]: It is used in line 19 because of the `std::function`

**Comment by Florisheinen1** [comment]
@emielsteerneman 
The whole reason why I want an intermediate packet being used, is to make sure our repositories are completely independent of what kind of simulator we use. The SSL protocol changes once in a while, and to have to then update the packets we use would mean we would have to update AI, networking and robothub, instead of just the simulator manager inside robothub that converts our intermediate packets into packets any simulator could understand. We could then even connect to different simulators simultaneously, without having to change stuff in AI for example.
And the grSim packets are a bit more complex due to the need for backwards compatibility, so to use them could cause some confusion to people who are new to this

**Comment by emielsteerneman** [comment]
lgtm. tested it and it works. feature request for also adding stuff like field width etc

**Pull request merged**